src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -118,14 +131,48 @@ public async Task StartContainerAsync(IExecutionContext executionContext, object
                 }
             }
 
-#if !OS_WINDOWS
-            executionContext.Debug($"Delete stale container networks from previous jobs");
-            int networkPruneExitCode = await _dockerManger.DockerNetworkPrune(executionContext);
-            if (networkPruneExitCode != 0)
+            foreach (var container in containers)
             {
-                executionContext.Warning($"Delete stale container networks failed, docker network prune fail with exit code {networkPruneExitCode}");
+                await StartContainerAsync(executionContext, container);
             }
-#endif
+        }
+
+        public async Task StopContainersAsync(IExecutionContext executionContext, object data)
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+
+            List<ContainerInfo> containers = data as List<ContainerInfo>;
+            ArgUtil.NotNull(containers, nameof(containers));
+
+            var uniqueNetworks = new HashSet<string>();
+            foreach (var container in containers)
+            {
+                await StopContainerAsync(executionContext, container);
+                if (!uniqueNetworks.Contains(container.ContainerNetwork) && !String.IsNullOrEmpty(container.ContainerNetwork))
+                {
+                    uniqueNetworks.Add(container.ContainerNetwork);
+                }
+            }
+            // Remove any unique network created for the job. There will usually be only one
+            foreach (var networkId in uniqueNetworks)
```

Comment:
i thought we only create on network, so why this is a list? :D

Reply:
Fixed